### PR TITLE
[TS] Migrate RadioButtonGroup tests to .tsx

### PR DESCRIPTION
**Pull request by gab0gomes**
#### What does this PR do?
* Migrates RadioButtonGroup tests to TS

#### Where should the reviewer start?
* Components => RadioButtonGroup

#### What testing has been done on this PR?
* Pre-written tests have been passed successfully.

#### How should this be manually tested?
* Ensure types in Nave are correct.

#### Any background context you want to provide?
* I have added a new entry on RadioButtonGroupProps option type to provide a way to use `data-testid` in tests files and to catch the use case where `...optionRest` exists;

#### What are the relevant issues?
#5599 

#### Screenshots (if appropriate)
image

#### Do the grommet docs need to be updated?
* No

#### Should this PR be mentioned in the release notes?
* No

#### Is this change backwards compatible or is it a breaking change?
*  Backwards compatible

### CODE REVIEW

File: src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.tsx [hunk 1]
```diff
@@ -200,74 +199,88 @@ describe('RadioButtonGroup', () => {
...
+        id: 'THREE',
+        value: '3',
+        label: 'radio button 3',
+      },
+    ];
+
     const onChange = jest.fn();
 
-    const { getByTestId } = render(
+    const { getByRole } = render(
```
Comment (jcfilben) [inline src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.tsx:222]:
```javascript
render(
      <Grommet>
        <RadioButtonGroup
          name="test"
          value="2"
          options={radioGroupOptions}
          onChange={onChange}
        />
      </Grommet>,
    );

    // Focus radio '2' button and simulate ArrowDown key
    // should result in selecting radio '3'
    const middleRadioBtn = screen.getByRole('radio', { name: 'radio button 2' });
    middleRadioBtn.focus();
    expect(middleRadioBtn).toHaveFocus();
    await waitFor(() => expect(middleRadioBtn).toHaveFocus(), {
      timeout: 200,
    });

    // focusing the radio button results in internal state update
    // so we wait (`act`) after focusing
    userEvent.type(middleRadioBtn, '{arrowDown}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 3' }),
      }),
    );

    // Focus radio '2' button and simulate ArrowUp key
    // should result in selecting radio '1'
    middleRadioBtn.focus();
    await waitFor(() => expect(middleRadioBtn).toHaveFocus(), {
      timeout: 200,
    });

    userEvent.type(middleRadioBtn, '{arrowUp}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 1' }),
      }),
    );
```

I think we can change the test to something like this. For what I can tell it is consistently passing. When you change the test to expect a different RadioButton it fails as expected.

**Comment by jcfilben** [comment]
I updated the test and merged with master. Unfortunately during that process it pulled in lint changes on the DataTable files. Note that the only code changes are in the `RadioButtonGroup-test.tsx` and `RadioButtonGroup-test.tsx.snap` files.

**Review by amandadupell — APPROVED** [review]
lgtm!

**Comment by britt6612** [comment]
This PR is now cleaned up 👍

**Comment by jcfilben** [comment]
I'm not sure why the "Works with keyboard" test is failing, locally it is passing for me even after merging with master

**Comment by halocline** [comment]
> I'm not sure why the "Works with keyboard" test is failing, locally it is passing for me even after merging master into the branch

If I run in watch mode `yarn test --watch`, set watch to only run `radiobuttongroup`, and then trigger tests runs by hitting `Enter` I get unpredictable results; sometimes it will pass, others it will fail.
Screen Shot 2021-12-15 at 10 48 44 AM
Screen Shot 2021-12-15 at 10 48 34 AM

One pattern I am noticing is that when the "Works with keyboard" test completes < ~300ms it fails; > ~300ms it will pass. ---> Investigating from this angle.

**Review by jcfilben — COMMENTED** [review]
Sorry for the long wait on this one. React testing library has changed a bit since the time this pull request was originally filed. I re-wrote the 'Works with keyboard' test with the latest testing library changes and it seems to be working well. Here is the updated test. I think if we change to this we should be good to go.
```javascript
test('Works with keyboard', async () => {
    const user = userEvent.setup();
    const radioGroupOptions = [
      {
        id: 'ONE',
        value: '1',
        label: 'radio button 1',
      },
      {
        id: 'TWO',
        value: '2',
        label: 'radio button 2',
      },
      {
        id: 'THREE',
        value: '3',
        label: 'radio button 3',
      },
    ];

    const onChange = jest.fn();

    render(
      <Grommet>
        <RadioButtonGroup
          name="test"
          value="2"
          options={radioGroupOptions}
          onChange={onChange}
        />
      </Grommet>,
    );

    const middleRadioBtn = screen.getByRole('radio', {
      name: 'radio button 2',
    });

    await user.type(middleRadioBtn, '{arrowDown}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 3' }),
      }),
    );

    await user.type(middleRadioBtn, '{arrowUp}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 1' }),
      }),
    );
  });
```

**Comment by jcfilben** [comment]
Sorry some lint changes got pulled in when I updated the branch with master

**Review by jcfilben — APPROVED** [review]

**Pull request merged**